### PR TITLE
refactor openshift_base.realize_data to run with parallelism

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-3416

### CODE REVIEW

**Review by Piojo — APPROVED** [review]
Minor nits, they can wait for a different iteration.

File: reconcile/openshift_base.py [hunk 1]
```diff
@@ -307,151 +307,167 @@ def check_unused_resource_types(ri):
...
+
+    # current items
+    for name, c_item in data['current'].items():
+        d_item = data['desired'].get(name)
+        if d_item is not None:
+            continue
 
-            if c_item.has_owner_reference():
+        if c_item.has_qontract_annotations():
+            if caller and c_item.caller != caller:
```
[the remark this answers is not in the document]
Comment (maorfr) [reply reconcile/openshift_base.py:414]: that would make the `elif` problematic.

File: reconcile/openshift_base.py [hunk 2]
```diff
@@ -307,151 +307,167 @@ def check_unused_resource_types(ri):
...
+                'cluster': cluster,
+                'namespace': namespace,
+                'kind': resource_type,
+                'name': name
+            }
+            actions.append(action)
+        except StatusCodeError as e:
+            ri.register_error()
+            msg = "[{}/{}] {}".format(cluster, namespace, str(e))
+            logging.error(msg)
```
[the remark this answers is not in the document]
Comment (maorfr) [reply reconcile/openshift_base.py:436]: the status code error would contain the full error message, which includes the required details.

File: reconcile/openshift_base.py [hunk 3]
```diff
@@ -307,151 +307,167 @@ def check_unused_resource_types(ri):
...
+        d_item = data['desired'].get(name)
+        if d_item is not None:
+            continue
 
-            if c_item.has_owner_reference():
+        if c_item.has_qontract_annotations():
+            if caller and c_item.caller != caller:
                 continue
+        elif not take_over:
+            continue
```
[the remark this answers is not in the document]
Comment (maorfr) [reply reconcile/openshift_base.py:423]: added a comment and debug logging

**Pull request merged**